lib/src/firebase_storage_mocks_base.dart

```diff
@@ -9,26 +9,26 @@ class MockFirebaseStorage extends Mock implements FirebaseStorage {
   final Map<String, Uint8List> storedDataMap = {};
 
   @override
-  StorageReference ref() {
+  Reference ref([String path]) {
     return MockStorageReference(this);
   }
 }
 
-class MockStorageUploadTask extends Mock implements StorageUploadTask {
-  final StorageReference reference;
+class MockStorageUploadTask extends Mock implements UploadTask {
+  final Reference reference;
 
   MockStorageUploadTask(this.reference);
 
   @override
-  Future<StorageTaskSnapshot> get onComplete =>
+  Future<TaskSnapshot> get onComplete =>
```

Comment:
The Dart Analyzer returns this warning:
> The getter doesn't override an inherited getter. Try updating this class to match the superclass, or removing the override annotation.

This is because onComplete has been removed:

> BREAKING: isCanceled, isComplete, isInProgress, isPaused and isSuccessful have now been removed.

Reply:
To make `MockStorageUploadTask` act like a `Future` and return a snapshot on `await`, you can use the following snippet. I tried it locally and it works well (provided you update the `bucket` and `path` getters):

```dart
class MockStorageUploadTask extends Mock implements UploadTask {
  final Future<TaskSnapshot> delegate;

  MockStorageUploadTask(reference)
      : delegate = Future.value(MockStorageTaskSnapshot(reference));

  @override
  Future<S> then<S>(FutureOr<S> Function(TaskSnapshot) onValue,
          {Function onError}) =>
      delegate.then(onValue, onError: onError);
}
```